<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [mod_python] ANNOUNCE: Mod_python 3.2.5 Beta
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mod_python%40modpython.org?Subject=%5Bmod_python%5D%20ANNOUNCE%3A%20Mod_python%203.2.5%20Beta&In-Reply-To=">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="019638.html">
   <LINK REL="Next"  HREF="019588.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[mod_python] ANNOUNCE: Mod_python 3.2.5 Beta</H1>
<table border=0 width="100%"><tr><td valign="top">
    <B>Gavin</B> 
    <A HREF="mailto:mod_python%40modpython.org?Subject=%5Bmod_python%5D%20ANNOUNCE%3A%20Mod_python%203.2.5%20Beta&In-Reply-To="
       TITLE="[mod_python] ANNOUNCE: Mod_python 3.2.5 Beta">gavin at sz.net.cn
       </A><BR>
    <I>Wed Nov 30 23:00:56 EST 2005</I>
    <P><UL>
        <LI>Previous message: <A HREF="019638.html">[mod_python] ANNOUNCE: Mod_python 3.2.5 Beta
</A></li>
        <LI>Next message: <A HREF="019588.html">[mod_python] leap of faith..
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#19639">[ date ]</a>
              <a href="thread.html#19639">[ thread ]</a>
              <a href="subject.html#19639">[ subject ]</a>
              <a href="author.html#19639">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Dear Scott,

The 3.2.5 Beta still have problem with Session, I tested it yestoday.

I am using mod_python3.2.5b(python2.4.1,apache2.0.55), I write a test program as 
following to test Session,but Failed. I have attached the Error message as following. 
This code works correctly in mod_python3.1.14(python2.3.5),I don't kown its why?


&lt;%
from mod_python import apache,Session

sid='1223'
secret=&quot;Abstfgsd&quot;
S=Session.Session(req,sid,secret,1800)
sid=S.id()
if S.is_new():
    S.save()
    req.write(sid)
else:
   S.load()
   req.write(sid)
%&gt;

&lt;%
%&gt;

The Error message in apache is as following:

[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp: Traceback (most recent call last):
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/site-pack
ages/mod_python/apache.py&quot;, line 299, in HandlerDispatch\n    result = object(req)
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/site-pack
ages/mod_python/psp.py&quot;, line 302, in handler\n    p.run()
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/site-pack
ages/mod_python/psp.py&quot;, line 213, in run\n    exec code in global_scope
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/data/HTTPD/htdocs/FeedBack/sid.m&quot;
, line 9, in ?\n    S.save()
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/site-pack
ages/mod_python/Session.py&quot;, line 209, in save\n    self.do_save(dict)
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/site-pack
ages/mod_python/Session.py&quot;, line 335, in do_save\n    dbm = self._get_dbm()
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/site-pack
ages/mod_python/Session.py&quot;, line 310, in _get_dbm\n    result = self._dbmtype.open(self._dbmfile, 'c')
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/anydbm.py
&quot;, line 83, in open\n    return mod.open(file, flag, mode)
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/dbhash.py
&quot;, line 16, in open\n    return bsddb.hashopen(file, flag, mode)
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/bsddb/__i
nit__.py&quot;, line 285, in hashopen\n    e = _openDBEnv()
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp:   File &quot;/usr/local/lib/python2.4/bsddb/__i
nit__.py&quot;, line 339, in _openDBEnv\n    e.open('.', db.DB_PRIVATE | db.DB_CREATE | db.DB_THREAD | db.DB_INIT_LOCK | db.DB_INIT_MPOO
L)
[Wed Nov 30 16:18:19 2005] [error] [client 202.104.96.163] PythonHandler mod_python.psp: DBError: (159940408, 'Unknown error 159940
408')


Sincerely,

Frank Ning



----- Original Message ----- 
From: &quot;Scott Chapman&quot; &lt;<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">scott_list at mischko.com</A>&gt;
To: &lt;<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">mod_python at modpython.org</A>&gt;
Sent: Thursday, December 01, 2005 11:04 AM
Subject: Re: [mod_python] ANNOUNCE: Mod_python 3.2.5 Beta


Does anyone have this available binary for Windows and Python 2.4.2?

Gregory (Grisha) Trubetskoy wrote:
&gt;<i> 
</I>&gt;<i> The Apache Software Foundation and The Apache HTTP Server Project are 
</I>&gt;<i> pleased to announce the 3.2.5 Beta release mod_python.
</I>&gt;<i> 
</I>&gt;<i> Version 3.2.5b of mod_python features several new functions and 
</I>&gt;<i> attributes providing better access to apache internals, file-based 
</I>&gt;<i> sessions and other session improvements, as well as many bug fixes and 
</I>&gt;<i> various performance and security improvements. A detailed description of 
</I>&gt;<i> the changes is available in Appendix A of the mod_python manual, also 
</I>&gt;<i> available here:
</I>&gt;<i> 
</I>&gt;<i> <A HREF="http://www.modpython.org/live/mod_python-3.2.5b/doc-html/node97.html">http://www.modpython.org/live/mod_python-3.2.5b/doc-html/node97.html</A>
</I>&gt;<i> 
</I>&gt;<i> Beta releases are NOT considered stable and usually contain bugs.
</I>&gt;<i> 
</I>&gt;<i> This release is intended to solicit widespread testing of the code. We 
</I>&gt;<i> strongly recommend that you try out your existing applications and 
</I>&gt;<i> experiment with new features in a non-production environment using this 
</I>&gt;<i> version and report any problems you may encounter so that they can be 
</I>&gt;<i> addressed before the final release.
</I>&gt;<i> 
</I>&gt;<i> Preferred method of reporting problems is the mod_python user list 
</I>&gt;<i> <A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">mod_python at modpython.org.</A>
</I>&gt;<i> 
</I>&gt;<i> Mod_python 3.2.5b is available for download from:
</I>&gt;<i> 
</I>&gt;<i> <A HREF="http://httpd.apache.org/modules/python-download.cgi">http://httpd.apache.org/modules/python-download.cgi</A>
</I>&gt;<i> 
</I>&gt;<i> For more information about mod_python visit <A HREF="http://www.modpython.org/">http://www.modpython.org/</A>
</I>&gt;<i> 
</I>&gt;<i> Regards,
</I>&gt;<i> 
</I>&gt;<i> Grisha Trubetskoy and the Apache mod_python team.
</I>&gt;<i> _______________________________________________
</I>&gt;<i> Mod_python mailing list
</I>&gt;<i> <A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">Mod_python at modpython.org</A>
</I>&gt;<i> <A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">http://mailman.modpython.org/mailman/listinfo/mod_python</A>
</I>&gt;<i> 
</I>&gt;<i> 
</I>
_______________________________________________
Mod_python mailing list
<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">Mod_python at modpython.org</A>
<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">http://mailman.modpython.org/mailman/listinfo/mod_python</A>

</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="019638.html">[mod_python] ANNOUNCE: Mod_python 3.2.5 Beta
</A></li>
	<LI>Next message: <A HREF="019588.html">[mod_python] leap of faith..
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#19639">[ date ]</a>
              <a href="thread.html#19639">[ thread ]</a>
              <a href="subject.html#19639">[ subject ]</a>
              <a href="author.html#19639">[ author ]</a>
         </LI>
       </UL>

</td>
<td align="right" valign="top">
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-42971867-2', 'modpython.org');
  ga('send', 'pageview');
</script>

</td>
</table>

<hr>
<a href="http://mailman.modpython.org/mailman/listinfo/mod_python">More information about the Mod_python
mailing list</a><br>
</body></html>
